doctr/models/recognition/sar.py

```diff
@@ -196,13 +192,21 @@ def __init__(
     def call(
         self,
         x: tf.Tensor,
+        labels: Optional[tf.sparse.SparseTensor] = None,
+        training: bool = False,
         **kwargs: Any,
     ) -> tf.Tensor:
 
         features = self.feat_extractor(x, **kwargs)
         pooled_features = tf.reduce_max(features, axis=1)  # vertical max pooling
         encoded = self.encoder(pooled_features, **kwargs)
-        decoded = self.decoder(features, encoded, **kwargs)
+        if training:
+            if labels:
+                decoded = self.decoder(features, encoded, labels, **kwargs)
+            else:
+                raise ValueError('Need to provide labels during training for teacher forcing')
+        else:
+            decoded = self.decoder(features, encoded, **kwargs)
```

Comment:
you need to pass down `training=training` to the decoder call if you do this

Reply:
why ? I think it is useless because I already checked inside the sar if we are in training mode